src/react/elements.js

```diff
@@ -109,6 +110,11 @@ function createPropsObject(
     // do we really need to create an object just for { children }
     if (children !== undefined) {
       let childrenObject = Create.ObjectCreate(realm, realm.intrinsics.ObjectPrototype);
+      // if we have default props, then we need to ensure the children
+      // have keys if they are an array as we're spreading them
+      if (defaultProps !== realm.intrinsics.undefined) {
+        children = applyBranchedLogicValue(realm, children);
```

Comment:
Is this fixing a bug? Which one?

I don't see this in the old code

Reply:
It was something missed off, not sure of any bugs, but this is the right logic to do after speaking with @sebmarkbage.

Reply:
I tried to come up with a repro, but couldn't make one that fails. If you can think of one, let me know. From memory, I think it's to do with key checks on children from props vs explicit children.

Reply:
This needs a test case then?